### PR TITLE
fix typescript error

### PR DESCRIPTION
**Pull request by yann300**

### CODE REVIEW

File: libs/remix-ws-templates/src/templates/ozerc20/scripts/ethers-lib.ts [hunk 1]
```diff
@@ -1,24 +1,27 @@
...
 
     const factory = new ethers.ContractFactory(metadata.abi, metadata.data.bytecode.object, signer);
 
-    let contract
-    if (from) {
-        contract = await factory.connect(from).deploy(...args);
-    } else {
-        contract = await factory.deploy(...args);
-    }    
+    let contract = await factory.deploy(...args)
```
Comment (Aniket-Engg) [inline libs/remix-ws-templates/src/templates/ozerc20/scripts/ethers-lib.ts:24]: `let` can be `const`

File: libs/remix-ws-templates/src/templates/ozerc721/scripts/ethers-lib.ts [hunk 2]
```diff
@@ -1,24 +1,27 @@
...
 
-export const deploy = async (contractName: string, args: Array<any>, from?: string): Promise<any> => {    
+/**
+ * Deploy the given contract
+ * @param {string} contractName name of the contract to deploy
+ * @param {Array<any>} args list of constructor' parameters
+ * @param {Number} acountIndex account index from the exposed account
+ * @return {Contract} deployed contract
+ */
+export const deploy = async (contractName: string, args: Array<any>, acountIndex?: number): Promise<ethers.Contract> => {
```
Comment (Aniket-Engg) [inline libs/remix-ws-templates/src/templates/ozerc721/scripts/ethers-lib.ts:10]: typo

File: libs/remix-ws-templates/src/templates/ozerc721/scripts/ethers-lib.ts [hunk 3]
```diff
@@ -1,24 +1,27 @@
 import { ethers } from 'ethers'
 
-export const deploy = async (contractName: string, args: Array<any>, from?: string): Promise<any> => {    
+/**
+ * Deploy the given contract
+ * @param {string} contractName name of the contract to deploy
+ * @param {Array<any>} args list of constructor' parameters
+ * @param {Number} acountIndex account index from the exposed account
```
Comment (Aniket-Engg) [inline libs/remix-ws-templates/src/templates/ozerc721/scripts/ethers-lib.ts:7]: typo

File: libs/remix-ws-templates/src/templates/ozerc721/scripts/web3-lib.ts [hunk 4]
```diff
@@ -1,8 +1,17 @@
 import Web3 from 'web3'
+import { Contract, ContractSendMethod, Options } from 'web3-eth-contract';
```
Comment (Aniket-Engg) [inline libs/remix-ws-templates/src/templates/ozerc721/scripts/web3-lib.ts:2]: semicolon

File: libs/remix-ws-templates/src/templates/ozerc20/scripts/web3-lib.ts [hunk 5]
```diff
@@ -12,14 +21,14 @@ export const deploy = async (contractName: string, args: Array<any>, from?: stri
 
     const accounts = await web3.eth.getAccounts()
 
-    let contract = new web3.eth.Contract(metadata.abi)
+    let contract: Contract  = new web3.eth.Contract(metadata.abi)
```
Comment (Aniket-Engg) [inline libs/remix-ws-templates/src/templates/ozerc20/scripts/web3-lib.ts:24]: `let` can be `const`

File: libs/remix-ws-templates/src/templates/ozerc20/scripts/web3-lib.ts [hunk 6]
```diff
@@ -12,14 +21,14 @@ export const deploy = async (contractName: string, args: Array<any>, from?: stri
...
 
-    contract = contract.deploy({
+    let contractSend: ContractSendMethod = contract.deploy({
```
Comment (Aniket-Engg) [inline libs/remix-ws-templates/src/templates/ozerc20/scripts/web3-lib.ts:26]: `let` can be `const`

File: libs/remix-ws-templates/src/templates/ozerc721/scripts/web3-lib.ts [hunk 7]
```diff
@@ -12,14 +21,14 @@ export const deploy = async (contractName: string, args: Array<any>, from?: stri
 
     const accounts = await web3.eth.getAccounts()
 
-    let contract = new web3.eth.Contract(metadata.abi)
+    let contract: Contract  = new web3.eth.Contract(metadata.abi)
 
-    contract = contract.deploy({
+    let contractSend: ContractSendMethod = contract.deploy({
```
Comment (Aniket-Engg) [inline libs/remix-ws-templates/src/templates/ozerc721/scripts/web3-lib.ts:26]: `let` can be `const`

File: libs/remix-ws-templates/src/templates/ozerc721/scripts/web3-lib.ts [hunk 8]
```diff
@@ -12,14 +21,14 @@ export const deploy = async (contractName: string, args: Array<any>, from?: stri
 
     const accounts = await web3.eth.getAccounts()
 
-    let contract = new web3.eth.Contract(metadata.abi)
+    let contract: Contract  = new web3.eth.Contract(metadata.abi)
```
Comment (Aniket-Engg) [inline libs/remix-ws-templates/src/templates/ozerc721/scripts/web3-lib.ts:24]: `let` can be `const`

File: libs/remix-ws-templates/src/templates/remixDefault/scripts/web3-lib.ts [hunk 9]
```diff
@@ -1,25 +1,34 @@
 import Web3 from 'web3'
+import { Contract, ContractSendMethod, Options } from 'web3-eth-contract';
```
Comment (Aniket-Engg) [inline libs/remix-ws-templates/src/templates/remixDefault/scripts/web3-lib.ts:2]: semicolon

File: libs/remix-ws-templates/src/templates/remixDefault/scripts/web3-lib.ts [hunk 10]
```diff
@@ -1,25 +1,34 @@
...
 
     const metadata = JSON.parse(await remix.call('fileManager', 'getFile', artifactsPath))
 
     const accounts = await web3.eth.getAccounts()
 
-    let contract = new web3.eth.Contract(metadata.abi)
+    let contract: Contract  = new web3.eth.Contract(metadata.abi)
 
-    contract = contract.deploy({
+    let contractSend: ContractSendMethod = contract.deploy({
```
Comment (Aniket-Engg) [inline libs/remix-ws-templates/src/templates/remixDefault/scripts/web3-lib.ts:26]: `let` can be `const`

File: libs/remix-ws-templates/src/templates/remixDefault/scripts/web3-lib.ts [hunk 11]
```diff
@@ -1,25 +1,34 @@
...
     // Make sure contract is compiled and artifacts are generated
-    const artifactsPath = `browser/contracts/artifacts/${contractName}.json` // Change this for different path
+    const artifactsPath = `browser/contracts/artifacts/${contractName}.json`
 
     const metadata = JSON.parse(await remix.call('fileManager', 'getFile', artifactsPath))
 
     const accounts = await web3.eth.getAccounts()
 
-    let contract = new web3.eth.Contract(metadata.abi)
+    let contract: Contract  = new web3.eth.Contract(metadata.abi)
```
Comment (Aniket-Engg) [inline libs/remix-ws-templates/src/templates/remixDefault/scripts/web3-lib.ts:24]: `let` can be `const`

File: libs/remix-ws-templates/src/templates/zeroxErc20/scripts/ethers-lib.ts [hunk 12]
```diff
@@ -1,24 +1,27 @@
 import { ethers } from 'ethers'
 
-export const deploy = async (contractName: string, args: Array<any>, from?: string): Promise<any> => {    
+/**
+ * Deploy the given contract
+ * @param {string} contractName name of the contract to deploy
+ * @param {Array<any>} args list of constructor' parameters
+ * @param {Number} acountIndex account index from the exposed account
```
Comment (Aniket-Engg) [inline libs/remix-ws-templates/src/templates/zeroxErc20/scripts/ethers-lib.ts:7]: typo

File: libs/remix-ws-templates/src/templates/zeroxErc20/scripts/ethers-lib.ts [hunk 13]
```diff
@@ -1,24 +1,27 @@
...
 
     const factory = new ethers.ContractFactory(metadata.abi, metadata.data.bytecode.object, signer);
 
-    let contract
-    if (from) {
-        contract = await factory.connect(from).deploy(...args);
-    } else {
-        contract = await factory.deploy(...args);
-    }    
+    let contract = await factory.deploy(...args)
```
Comment (Aniket-Engg) [inline libs/remix-ws-templates/src/templates/zeroxErc20/scripts/ethers-lib.ts:24]: `let` can be `const`

File: libs/remix-ws-templates/src/templates/zeroxErc20/scripts/ethers-lib.ts [hunk 14]
```diff
@@ -1,24 +1,27 @@
...
-    const artifactsPath = `browser/contracts/artifacts/${contractName}.json`
+    const artifactsPath = `browser/contracts/artifacts/${contractName}.json` // Change this for different path
 
     const metadata = JSON.parse(await remix.call('fileManager', 'getFile', artifactsPath))
     // 'web3Provider' is a remix global variable object
-    const signer = (new ethers.providers.Web3Provider(web3Provider)).getSigner()
+    
+    const signer = (new ethers.providers.Web3Provider(web3Provider)).getSigner(acountIndex)
 
     const factory = new ethers.ContractFactory(metadata.abi, metadata.data.bytecode.object, signer);
```
Comment (Aniket-Engg) [inline libs/remix-ws-templates/src/templates/zeroxErc20/scripts/ethers-lib.ts:22]: semicolon

File: libs/remix-ws-templates/src/templates/zeroxErc20/scripts/web3-lib.ts [hunk 15]
```diff
@@ -1,8 +1,17 @@
 import Web3 from 'web3'
+import { Contract, ContractSendMethod, Options } from 'web3-eth-contract';
```
Comment (Aniket-Engg) [inline libs/remix-ws-templates/src/templates/zeroxErc20/scripts/web3-lib.ts:2]: semicolon

File: libs/remix-ws-templates/src/templates/zeroxErc20/scripts/web3-lib.ts [hunk 16]
```diff
@@ -12,14 +21,14 @@ export const deploy = async (contractName: string, args: Array<any>, from?: stri
 
     const accounts = await web3.eth.getAccounts()
 
-    let contract = new web3.eth.Contract(metadata.abi)
+    let contract: Contract  = new web3.eth.Contract(metadata.abi)
 
-    contract = contract.deploy({
+    let contractSend: ContractSendMethod = contract.deploy({
```
Comment (Aniket-Engg) [inline libs/remix-ws-templates/src/templates/zeroxErc20/scripts/web3-lib.ts:26]: `let` can be `const`

File: libs/remix-ws-templates/src/templates/zeroxErc20/scripts/web3-lib.ts [hunk 17]
```diff
@@ -12,14 +21,14 @@ export const deploy = async (contractName: string, args: Array<any>, from?: stri
 
     const accounts = await web3.eth.getAccounts()
 
-    let contract = new web3.eth.Contract(metadata.abi)
+    let contract: Contract  = new web3.eth.Contract(metadata.abi)
```
Comment (Aniket-Engg) [inline libs/remix-ws-templates/src/templates/zeroxErc20/scripts/web3-lib.ts:24]: `let` can be `const`

**Review by bunsenstraat — APPROVED** [review]

**Pull request merged**